test cases/common/183 as-needed/libA.cpp

```diff
@@ -0,0 +1 @@
+bool linked = false;
```

Comment:
It is probably fine now but I would recommend wrapping it in a namespace to avoid any weird name collision.
`namespace meson_test_as_needed` or something else unique enough.